src/all/cosplaytele/src/eu/kanade/tachiyomi/extension/all/cosplaytele/CosplayTele.kt

```diff
@@ -0,0 +1,227 @@
+        val genres = getTags(document)
+        manga.genre = genres.joinToString(", ")
+        manga.status = SManga.COMPLETED
+        return manga
+    }
+
+    private fun getTags(document: Element): List<String> {
+        val pattern = """.*/(tag|category)/.*""".toRegex()
+        return document.select("#main a").filter { a -> pattern.matches(a.attr("href")) }.map { a ->
+            val link = a.attr("href").split(".com/")[1]
+            val tag = a.text()
+            Log.d("app.mihon.debug", "here $tag/$link")
+            if (tag.isNotEmpty()) {
+                this.categories[tag] = link
+            }
+            tag
+        }
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val chapter = SChapter.create()
+        chapter.setUrlWithoutDomain(element.select("link[rel=\"canonical\"]").attr("href"))
+        chapter.chapter_number = -2f
+        chapter.name = "Gallery"
+        chapter.date_upload = getDate(element.select("time.updated").attr("datetime"))
+        return chapter
+    }
+
+    override fun chapterListSelector() = "html"
+
+    // Pages
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = mutableListOf<Page>()
+        document.select("noscript").remove()
+        document.select(".gallery-item img").forEachIndexed { i, it ->
+            val itUrl = it.imgSrc
+            pages.add(Page(i, itUrl, itUrl))
+        }
+        return pages
+    }
+
+    override fun imageUrlParse(document: Document): String =
+        throw UnsupportedOperationException()
+
+    // Filters
+    override fun getFilterList(): FilterList {
+        val filters = mutableListOf<Filter<*>>(
+            Filter.Header("NOTE: Only one filter will be applied!"),
+            Filter.Separator(),
+            UriPartFilter("Category", categories.entries.toTypedArray()),
+        )
+
+        if (filtersState == FilterState.Unfetched) {
+            filters.add(1, Filter.Header("Use 'reset' to load all filters"))
+        }
+        return FilterList(filters)
+    }
+
+    open class UriPartFilter(
+        displayName: String,
+        private val valuePair: Array<MutableMap.MutableEntry<String, String>>,
+    ) : Filter.Select<String>(displayName, valuePair.map { it.key }.toTypedArray()) {
+        fun toUriPart() = valuePair[state].value
+    }
+
+    private var categories = mutableMapOf(
+        Pair("All", ""),
+        Pair("Cosplay Nude", "category/nude"),
+        Pair("Cosplay Ero", "category/no-nude"),
+        Pair("Cosplay", "category/cosplay"),
+    )
+
+    private var filtersState = FilterState.Unfetched
+    private var filterAttempts = 0
+
+    private enum class FilterState {
+        Fetching, Fetched, Unfetched
+    }
+
+    private val scope = CoroutineScope(Dispatchers.IO)
+
+    protected fun launchIO(block: suspend () -> Unit) = scope.launch { block() }
+
+//
+    private suspend fun fetchFilters() {
```

Comment:
```suggestion
    private fun fetchFilters() {
```
you are not using suspend in the function so no need